### PR TITLE
Fix magic wall and wild growth rune bug

### LINKED ISSUE #507: Magic walls and wild growth disappear when someone steps over it
### What happened?

magic walls and wild growth disappear if a player or monster walks through them in open pvp and pvp enforced.
magic walls does not disappear in non pvp but monsters and players can walk through them.

### How to Reproduce?

x

### Relevant log output

```shell
x
```

### Attach any relevant file

local combat = Combat()
combat:setParameter(COMBAT_PARAM_DISTANCEEFFECT, CONST_ANI_EARTH)

function onCreateMagicWall(creature, tile)
	local item = Game.createItem(Game.getWorldType() == WORLD_TYPE_NO_PVP and ITEM_WILDGROWTH_SAFE or ITEM_WILDGROWTH, 1, tile)
	item:setAttribute(ITEM_ATTRIBUTE_DURATION, math.random(38000, 45000))
end

combat:setCallback(CALLBACK_PARAM_TARGETTILE, "onCreateMagicWall")

local spell = Spell("rune")
function spell.onCastSpell(creature, variant, isHotkey)
    return combat:execute(creature, variant)
end

spell:name("Wild Growth Rune")
spell:group("attack")
spell:id(94)
spell:cooldown(2 * 1000)
spell:groupCooldown(2 * 1000)
spell:level(27)
spell:magicLevel(8)
spell:runeId(3156)
spell:charges(2)
spell:isBlocking(true, true)
spell:allowFarUse(true)
spell:vocation("druid;true", "elder druid;true")
spell:register()

### Version

12.x (Default)

### What OS are you seeing the problem on?

Windows 10

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by dudantas**
Working with: https://github.com/opentibiabr/canary/pull/388

Fixed runes (bug related to wrong flag "blocking")

Created new function for set duration

```cpp
item:setDuration(minDurationRange, maxDurationRange = 0, decayid = 0, showDuration = true)
```

### CODE REVIEW

**Comment by Alissonlv123** [comment]
Aprovado Rodou Tranquilo E funcionando perfeitamente!

**Review by Alissonlv123 — APPROVED** [review]

File: data/scripts/spells/runes/wild_growth.lua [hunk 1]
```diff
@@ -1,16 +1,21 @@
-local combat = Combat()
-combat:setParameter(COMBAT_PARAM_DISTANCEEFFECT, CONST_ANI_EARTH)
-
-function onCreateMagicWall(creature, tile)
-	local item = Game.createItem(Game.getWorldType() == WORLD_TYPE_NO_PVP and ITEM_WILDGROWTH_SAFE or ITEM_WILDGROWTH, 1, tile)
-	item:setAttribute(ITEM_ATTRIBUTE_DURATION, math.random(38000, 45000))
+function onCreateWildGrowth(creature, tile)
+	local wildGrowth
+	if Game.getWorldType() == WORLD_TYPE_NO_PVP then
+		wildGrowth = ITEM_MAGICWALL_SAFE
```
Comment (guispiller) [inline data/scripts/spells/runes/wild_growth.lua]: wildGrowth = ITEM_WILDGROWTH_SAFE

File: data/scripts/spells/runes/wild_growth.lua [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
+	else
+		wildGrowth = ITEM_MAGICWALL
```
Comment (guispiller) [inline data/scripts/spells/runes/wild_growth.lua]: wildGrowth = ITEM_WILDGROWTH

File: data/scripts/spells/runes/wild_growth.lua [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
+	end
+	local item = Game.createItem(wildGrowth, 1, tile)
+	item:setDuration(30, 60)
 end
 
-combat:setCallback(CALLBACK_PARAM_TARGETTILE, "onCreateMagicWall")
+local combat = Combat()
+combat:setParameter(COMBAT_PARAM_DISTANCEEFFECT, CONST_ANI_EARTH)
```
Comment (guispiller) [inline data/scripts/spells/runes/wild_growth.lua]: combat:setParameter(COMBAT_PARAM_DISTANCEEFFECT, CONST_ANI_ENERGY)

**Review by guispiller — CHANGES_REQUESTED** [review]

**Review by guispiller — APPROVED** [review]

**Review by omeranha — APPROVED** [review]

**Review by majestyotbr — APPROVED** [review]

**Pull request merged**